doc/sphinx/proofs/writing-proofs/reasoning-inductives.rst

```diff
@@ -5,269 +5,282 @@ Reasoning with inductive types
+     the tactic to.  See :ref:`Occurrence clauses <occurrenceclauses>`.
+     If it occurs in the :n:`@induction_principle`, then
+     there can only be one :n:`@induction_clause`, which can't have its
+     own :n:`@occurrences` clause.
+
+   :n:`@induction_principle`
+     Makes the tactic equivalent to
+     :tacn:`induction` :n:`{+, @induction_clause } @induction_principle`.
+
+   .. tacn:: edestruct {+, @induction_clause } {? @induction_principle }
+
+      Equivalent to :n:`destruct @term` except that it does not
       fail if the instance of a dependent premises of the type
-      of :token:`term` is not inferable. Instead, the unresolved instances
+      of :token:`term` is not inferrable. Instead, the unresolved instances
       are left as existential variables to be inferred later, in the same way
-      as :tacn:`eapply` does.
-
-   .. tacv:: destruct @term using @term {? with @bindings }
-
-      This is synonym of :n:`induction @term using @term {? with @bindings }`.
-
-   .. tacv:: destruct @term in @goal_occurrences
-
-      This syntax is used for selecting which occurrences of :token:`term`
-      the case analysis has to be done on. The :n:`in @goal_occurrences`
-      clause is an occurrence clause whose syntax and behavior is described
-      in :ref:`occurrences sets <occurrencessets>`.
+      they are in :tacn:`eapply`.
 
-   .. tacv:: destruct @term {? with @bindings } {? as @or_and_intropattern_loc } {? eqn:@naming_intropattern } {? using @term {? with @bindings } } {? in @goal_occurrences }
-             edestruct @term {? with @bindings } {? as @or_and_intropattern_loc } {? eqn:@naming_intropattern } {? using @term {? with @bindings } } {? in @goal_occurrences }
+.. tacn:: case {+, @induction_clause } {? @induction_principle }
 
-      These are the general forms of :tacn:`destruct` and :tacn:`edestruct`.
-      They combine the effects of the ``with``, ``as``, ``eqn:``, ``using``,
-      and ``in`` clauses.
+   .. todo PR: clear as mud? (below) the syntax accepts more parameters than elim
+      not sure what to say here
 
-.. tacn:: case @term
-   :name: case
+   An older, more basic tactic to perform case analysis without
+   recursion.  We recommend using :tacn:`destruct` instead where possible.
+   `case` only modifies the goal; it does not modify the :term:`local context`.
 
-   The tactic :n:`case` is a more basic tactic to perform case analysis without
-   recursion. It behaves as :n:`elim @term` but using a case-analysis
-   elimination principle and not a recursive one.
+   .. tacn:: ecase {+, @induction_clause } {? @induction_principle }
 
-.. tacv:: case @term with @bindings
+      If the type of :n:`@one_term` (in :n:`@induction_arg`) has
+      :term:`dependent premises <dependent premise>`, or dependent premises
+      whose values are not inferrable from the :n:`with @bindings` clause,
+      :n:`ecase` turns them into existential variables to be resolved later on.
```

Comment:
This explanation is much better than the one for `edestruct`. Can you copy it over there?